### PR TITLE
replace master with main

### LINKED ISSUE #1080: fix broken links in markdown
The default branches across exercism have been renamed from `master` to `main`.
Many links in this repository refer to `master` branch.

See https://github.com/exercism/rust/tree/69e0e7dbfef8b239e69c22bb6e1c7681e092d85b/exercises/pig-latin for one example.

# goal
- replace `master` with `main` in concept exercise's README.md

### PR DESCRIPTION
**Pull request by coriolinus**
Just ran `rg master --files-with-matches | xargs sed -i 's/master/main/'` to get the bulk of the work; the problem now is to review and ensure we eliminate any false positives/negatives.

Closes #1080.

### CODE REVIEW

File: exercises/macros/example.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Ignoring the README's injunction, this is heavily based on the implementation from maplit.
-// Original source is at https://github.com/bluss/maplit/blob/master/src/lib.rs#L27-L60
+// Original source is at https://github.com/bluss/maplit/blob/main/src/lib.rs#L27-L60
```
Comment (coriolinus) [inline exercises/macros/example.rs:2]:
```suggestion
// Original source is at https://github.com/bluss/maplit/blob/master/src/lib.rs#L27-L60
```

`bluss/maplit` retains `master` as primary branch; checked now.

File: exercises/pig-latin/README.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ If you want to know more about Exercism, take a look at the [contribution guide]
 
 ## Source
 
-The Pig Latin exercise at Test First Teaching by Ultrasaurus [https://github.com/ultrasaurus/test-first-teaching/blob/master/learn_ruby/pig_latin/](https://github.com/ultrasaurus/test-first-teaching/blob/master/learn_ruby/pig_latin/)
+The Pig Latin exercise at Test First Teaching by Ultrasaurus [https://github.com/ultrasaurus/test-first-teaching/blob/main/learn_ruby/pig_latin/](https://github.com/ultrasaurus/test-first-teaching/blob/main/learn_ruby/pig_latin/)
```
Comment (coriolinus) [inline exercises/pig-latin/README.md:95]:
```suggestion
The Pig Latin exercise at Test First Teaching by Ultrasaurus <https://github.com/ultrasaurus/test-first-teaching/blob/master/learn_ruby/pig_latin/>
```

`ultrasaurus/test-first-teaching` still uses `master` as primary branch; checked now.

**Comment by efx** [comment]
@coriolinus you are too fast! nice work. I just started this incidentally in https://github.com/exercism/rust/tree/fix-1054.
I like the batch replace better.

**Review by efx — APPROVED** [review]

**Comment by efx** [comment]
I'll document this after @exercism/v3's work is merged here, but Rust maintainers have decided that the approver should *not* merge the work. Instead, the reviewer approves and leaves the merge to the contributor.

**Comment by coriolinus** [comment]
> the reviewer approves and leaves the merge to the contributor. 

Nuance: in cases where the contributor has commit permissions. In cases where the contributor cannot merge their own work, the approver will.

**Pull request merged**